Updated README

janettech opened the pull request:
Hello,

I have updated the README of Gaphas. I have reviewed and edited the file for :
1. Spelling and grammar
2. Edited the hyperlinks for virtual env

I do have comments on this line--
The constraint solver can also be used to keep constraints
contained within the item true, for example, to make sure a box maintains its
rectangular shape. (the word "true" is confusing).

danyeaw commented on README.md:
```diff
@@ -736,8 +739,9 @@ Move the item and lock the position of any element that's located at the same
-- view (`gaphas.view.View`): The view with guides to to use for move
-coordinates.
+- view (`gaphas.view.View`): The view with guides to use for move
+  
```
Why was an extra return added here and on line 762?

janettech replied:
Done

danyeaw commented on README.md:
```diff
@@ -835,20 +840,21 @@ rect = Rectangle(1, 1, 5, 5)
-- single (bool): Schedules the decorated function to only be called a signle
-time.
+- single (bool): Schedules the decorated function to be called only a single
+  
```
Extra return?

janettech replied:
Done

danyeaw commented on README.md:
```diff
@@ -879,33 +887,43 @@ def a(self, x=0):
 Thanks goes to these wonderful people ([emoji key](https://github.com/kentcdodds/all-contributors#emoji-key)):
 
 <!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
+
```
This section is autogenerated by the all-contributors tool, can we undo the modification?

janettech replied:
I didn't touch these lines. Not sure whats the issue. Can you check again and let me know what needs to be done. I just edited the summary lines now as there were line breaks.

danyeaw commented on README.md:
```diff
@@ -879,33 +887,43 @@ def a(self, x=0):
-4.  Send a pull request and bug the maintainers until it gets merged and
-    published. :smile:
+1. Check for open issues or open a fresh issue to start a discussion
+   
```
Why were the extra lines added in this section?

janettech replied:
Removed it.

danyeaw commented:
> The constraint solver can also be used to keep constraints contained within the item true, for example, to make sure a box maintains it rectangular shape. (the word "true" is confusing).

What if we use the term [satisfied](https://en.wikipedia.org/wiki/Constraint_satisfaction) instead of true? Would that be more clear for you?

janettech commented:
Hello Dan,

Thanks. I did have a feeling I did mess up with the indentation. Which markup editor did you use? I used MarkText and the content appearance was not the same as the README here. 

Let me know if you prefer a particular one and I will install that to edit your review comments.

danyeaw commented:
@janettech If Marktext isn't working, I would recommend a code editor like VSCode. There is a [tutorial](https://sciwiki.fredhutch.org/compdemos/vscode_markdown_howto/) about using Markdown with VSCode. Jump on Gitter if you are having any issues.

danyeaw commented:
@janettech How are things going?

janettech commented on README.md:
```diff
@@ -139,65 +143,63 @@ create_canvas(c, "Simple Gaphas App")
 depends on the case at hand, but most often involves the creation of some
 sort of constraint between the Handle and the item it is connecting to (see
-doc/ports.txt).
+doc/ports.rst).
```
Yes, this can be done.

janettech commented on README.md:
```diff
@@ -223,105 +225,96 @@ item, etc.
-tool has grabbed the focus. This normally happens when the user presses a mouse
-button. Once this happens, the tool requests a "grab" and all events, like
-motion or button release, are sent directly to the focused tool.
+- ToolChain - Delegates to a set of individual tools and keeps track of which tool has grabbed the focus. This normally happens when the user presses a mouse button. Once this happens, the tool requests a "grab" and all events, like motion or button release, are sent directly to the focused tool.
```
I have installed VScode and I see that they are correctly placed in the editor. I am not sure if I am missing anything here.

janettech commented:
Hi Dan,

I have updated the changes. I have a doubt:

Now I have added to edit your review comments. I have clicked on the pull request again. Are you able to see the modified file? Still learning ....:)

Thanks!

danyeaw commented:
@janettech This looks great, thanks so much for you contribution, I really appreciate it! :heart: :tada:

danyeaw commented:
@allcontributors please add @janettech for doc

amolenaar commented:
Thanks @janettech! I love how you improved the readme. :)

janettech commented:
Thanks @danyeaw and @amolenaar for your kind words. Glad I could do a small bit in your project :)
